### ISSUE TITLE
Emulated admin.initializeApp() does not return a fully mocked FirebaseApp

### ISSUE THREAD

**Issue by samtstern** [firebase/firebase-tools#1454]
### [REQUIRED] Environment info

**firebase-tools:** 7.0.2

**Platform:** gLinux

### [REQUIRED] Test case

Discovered while investigating #1451 

### [REQUIRED] Steps to reproduce

This works:
```js
const functions = require("firebase-functions");
const admin = require('firebase-admin');

admin.initializeApp();
const firestore = admin.firestore();

exports.writeToFirestore = functions.https.onRequest(async (req, res) => {
    const ref = firestore.doc("info/test");
    await ref.set({ hello: (new Date().toISOString()) });
    ref.get().then((snap) => {
        res.json({ data: snap.data() });
    });
});
```

This does not:
```js
const functions = require("firebase-functions");
const admin = require('firebase-admin');

const app = admin.initializeApp();
const firestore = app.firestore();

exports.writeToFirestore = functions.https.onRequest(async (req, res) => {
    const ref = firestore.doc("info/test");
    await ref.set({ hello: (new Date().toISOString()) });
    ref.get().then((snap) => {
        res.json({ data: snap.data() });
    });
});
```

### [REQUIRED] Expected behavior

The two above are equivalent.

### [REQUIRED] Actual behavior

When using the `firestore` instance from the `app` the first request fails with:
```
Error: The incoming JSON object does not contain a client_email field
```

**Comment by samtstern** [comment]
cc @noelmansour

**Comment by samtstern** [comment]
OK I figured out what we need to do.

`admin.firestore()` is just a shortcut for `admin.app.firestore()` so we need to be mocking the default app rather than the `admin` module:
https://github.com/firebase/firebase-admin-node/blob/1a7f722d2f4a9d918a29dbf6f088f0752cb74391/src/firebase-namespace.ts#L374

**Comment by samtstern** [comment]
Fixed by #1459
